guide/training_with_augmentations.ipynb

```diff
@@ -0,0 +1,360 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nobrainer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Get sample features and labels\n",
+    "\n",
+    "We use 9 pairs of volumes for training and 1 pair of volumes for evaulation. Many more volumes would be required to train a model for any useful purpose."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "csv_of_filepaths = nobrainer.utils.get_data()\n",
+    "filepaths = nobrainer.io.read_csv(csv_of_filepaths)\n",
+    "\n",
+    "train_paths = filepaths[:9]\n",
+    "evaluate_paths = filepaths[9:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Convert medical images to TFRecords\n",
+    "\n",
+    "Remember how many full volumes are in the TFRecords files. This will be necessary to know how many steps are in on training epoch. The default training method needs to know this number, because Datasets don't always know how many items they contain."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Verify that all volumes have the same shape and that labels are integer-ish.\n",
+    "\n",
+    "invalid = nobrainer.io.verify_features_labels(train_paths)\n",
+    "assert not invalid\n",
+    "\n",
+    "invalid = nobrainer.io.verify_features_labels(evaluate_paths)\n",
+    "assert not invalid"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir -p data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Convert training and evaluation data to TFRecords.\n",
+    "\n",
+    "nobrainer.tfrecord.write(\n",
+    "    features_labels=train_paths,\n",
+    "    filename_template='data/data-train_shard-{shard:03d}.tfrec',\n",
+    "    examples_per_shard=3)\n",
+    "\n",
+    "nobrainer.tfrecord.write(\n",
+    "    features_labels=evaluate_paths,\n",
+    "    filename_template='data/data-evaluate_shard-{shard:03d}.tfrec',\n",
+    "    examples_per_shard=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!ls data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create Datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_classes = 1\n",
+    "batch_size = 2\n",
+    "volume_shape = (256, 256, 256)\n",
+    "block_shape = (128, 128, 128)\n",
```

Comment:
I tried this notebook in collab. it does not go through with block size 128 because of RAM limitation but it works with 64 block size.

Reply:
aah! I tried on my local machine... yes let me put that to 32/64 maybe ...